tap_github.py

```diff
@@ -373,43 +365,47 @@ def do_sync(config, state, catalog):
     selected_stream_ids = get_selected_streams(catalog)
     validate_dependencies(selected_stream_ids)
 
-    for stream in catalog['streams']:
-        stream_id = stream['tap_stream_id']
-        stream_schema = stream['schema']
-        mdata = stream['metadata']
-
-        # if it is a "sub_stream", it will be sync'd by its parent
-        if not SYNC_FUNCTIONS.get(stream_id):
-            continue
-
-        # if stream is selected, write schema and sync
-        if stream_id in selected_stream_ids:
-            singer.write_schema(stream_id, stream_schema,stream['key_properties'])
-
-            # get sync function and any sub streams
-            sync_func = SYNC_FUNCTIONS[stream_id]
-            sub_stream_ids = SUB_STREAMS.get(stream_id, None)
-
-            # sync stream
-            if not sub_stream_ids:
-                state = sync_func(stream_schema, config, state, mdata)
-
-            # handle streams with sub streams
-            else:
-                stream_schemas = {stream_id: stream_schema}
-
-                # get and write selected sub stream schemas
-                for sub_stream_id in sub_stream_ids:
-                    if sub_stream_id in selected_stream_ids:
-                        sub_stream = get_stream_from_catalog(sub_stream_id, catalog)
-                        stream_schemas[sub_stream_id] = sub_stream['schema']
-                        singer.write_schema(sub_stream_id, sub_stream['schema'],
-                                            sub_stream['key_properties'])
-
-                # sync stream and it's sub streams
-                state = sync_func(stream_schemas, config, state, mdata)
-
-            singer.write_state(state)
+    repositories = list(filter(None, config['repository'].split(' ')))
```

Comment:
I think it would be better to rename the config key to `repositories` and have `repository` as an optional config that we could fall back to (and not split) if and only if `repositories` is not set.

Reply:
Right. This misnomer pained me a little bit but less so than the idea of flat-out changing it and causing a breaking change for anybody who already had a working configuration. Your suggesting is appealing so that it is more properly named but I don't love the idea of having a config property around that isn't really needed.

Would it make sense to introduce `repositories` now, and then remove `repository` in the future when a major version bump is more justified for the tap?

Reply:
> Would it make sense to introduce repositories now, and then remove repository in the future when a major version bump is more justified for the tap?

That is precisely the path forward that I would suggest. Support the old style (translating it to the new style on the fly with a log message) but prefer the new style. Then eventually we can major version bump it (definitely no need to do that now).